client/client.go

```diff
@@ -29,6 +30,7 @@ func NewClient(
 	}
 	return Client{
 		Client: c,
+		encCfg: encCfg,
```

Comment:
we only need `Codec`. 
```suggestion
		codec: codec.Codec,
```